testbed/correctnesstests/traces/correctness_test.go

```diff
@@ -92,3 +92,63 @@ func testWithTracingGoldenDataset(
 
 	tc.ValidateData()
 }
+
+func TestSporadicGoldenDataset(t *testing.T) {
```

Comment:
Will you add similar tests for metrics and logs?

Reply:
I will, but it would be repetitive content if I add it int this PR.
That's because, for metrics' correctness tests, we don't use our [validator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/testbed/testbed/validator.go) for metrics tests.
We rather create a new one.
In my opinion, we should keep one validator and use that for trace/metrics/log.
I will work on it after this PR.

Talking about logs, we don't have correctness tests for logs at all. It shouldn't be that difficult to add new ones.
I'd like to keep this in a different PR.

Reply:
and we don't use `MockBackend` in metrics correctness tests as well. We create a new `testHarness`. Which acts as both, a validator/backend.

Reply:
Ok, sounds good